### PR TITLE
Humdrum triangle: 2FA form for new auth flow

### PR DESCRIPTION
**Pull request by oliviacpu**
## Links
* Remix link: https://humdrum-triangle.glitch.me/
* Issue-tracker link: https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/268
* Specs/doc links: https://gallery.io/projects/MCHbtQVoQ2HCZZpaEieUFqhj/files/MCEJu8Y2hyDScckl30Wc2y4V2c6qLxWO8h0

## Changes:
* This changes the 2FA page to match the new designs, and adds a 2FA challenge to the login page

## How To Test:
* Enable "user passwords" and 2FA for your account, then go to https://humdrum-triangle.glitch.me/ and try to log in. You should be required to enter your 2FA code.

## Things left to do before deploying:
- [ ] I think after we deploy will need to hook this up to the login page after Sarah's username/password is merged.

### CODE REVIEW

File: src/components/layout/sign-in-layout.js [hunk 1]
```diff
@@ -2,146 +2,138 @@ import React, { useState } from 'react';
...
+                <OverlaySection type="info">
+                  <OverlayTitle goBack={goBack}>Magic Code</OverlayTitle>
+                </OverlaySection>
+                <OverlaySection type="actions">
+                  <UseMagicCode
+                    emailAddress={state.emailAddress}
+                    onTwoFactorChallenge={(initialToken) => {
+                      setState({ ...state, initialToken });
+                      setPage('2fa');
+                    }}
```
Comment (GregWeil) [inline src/components/layout/sign-in-layout.js]: This prop is called 'showTwoFactorPage', so it ends up falling to the try catch and showing the 'bad code' error.

File: src/presenters/pages/login/two-factor-code.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const TwoFactorCodePage = ({ initialToken, onSuccess }) => (
+  <AuthLayout>
+    <Overlay className={styles.overlay}>
+      <OverlaySection type="info">
+        <OverlayTitle>Two Factor Authentication</OverlayTitle>
+      </OverlaySection>
+      <OverlaySection type="actions">
+        <Notification type="success" persistent>
+          Almost Done
+        </Notification>
```
Comment (GregWeil) [inline src/presenters/pages/login/two-factor-code.js]: The TwoFactorForm also puts an 'Almost done' notification at the top, so the 2fa page has two in a row

**Comment by oliviacpu** [comment]
@GregWeil I do not know what the plan for /signin is now. I think that is the oauth page that lets people use their glitch accounts for the support forums. We'll have to talk about it with some other people/teams.

File: src/components/containers/preview-container.styl [hunk 3]
```diff
@@ -13,7 +13,3 @@
   flex: 1 0 auto
   font-size: 16px
   font-weight: bold
-
-dialog
-  top: 25% !important
```
Comment (sarahzinger) [inline src/components/containers/preview-container.styl]: 🙌

File: src/components/layout/sign-in-layout.js [hunk 4]
```diff
@@ -2,146 +2,138 @@ import React, { useState } from 'react';
...
-  const [page, setPage] = useState('main');
-  const [email, setEmail] = useState();
-  const showMainPage = () => setPage('main');
-  const showGetCodePage = () => setPage('getCode');
-  const showUseCodePage = () => setPage('useCode');
-  const showCreateAccountPage = () => setPage('createAccount');
+const MagicHat = () => <Image width={92} src="https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fmagic-link.svg?v=1568658265397" alt="" />;
+const Thumbprint = () => <Image width={92} src="https://cdn.glitch.com/02863ac1-a499-4a41-ac9c-41792950000f%2F2fa.svg?v=1568309705782" alt="" />;
+const BrainFriends = () => (
+  <Image width={92} src="https://cdn.glitch.com/0aa2fffe-82eb-4b72-a5e9-444d4b7ce805%2Fbrainfriends.svg?v=1568814632916" alt="" />
```
Comment (sarahzinger) [inline src/components/layout/sign-in-layout.js]: we can get rid of brain friends now if we want! but also if we want that in another pr that's good to me!

File: src/presenters/pages/login/two-factor-code.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Login page for Glitch VS Code Plugin
```
Comment (sarahzinger) [inline src/presenters/pages/login/two-factor-code.js]: is this just for the vs code plugin or are we planning on using this in multiple places? (apologies i forget!)

File: src/components/sign-in/use-magic-code.js [hunk 6]
```diff
@@ -35,8 +35,9 @@ const UseMagicCode = ({ emailAddress }) => {
     try {
       const { data } = await api.post(`/auth/email/${code}`);
       if (data.tfaToken) {
-        // TODO: Support 2FA here
-        setStatus('error');
+        showTwoFactorPage();
+        // TODO Error? Maybe not?
+        // setStatus('error');
```
[the remark this answers is not in the document]
Comment (oliviacpu) [reply src/components/sign-in/use-magic-code.js]: It gets caught below, so I think that was just an out of date TODO? Removing it!

**Pull request merged**